src/mame/drivers/jpmsys5.cpp

```diff
@@ -200,37 +201,197 @@ void jpmsys5v_state::rombank_w(uint16_t data)
 	m_rombank->set_entry(data & 0x1f);
 }
 
-uint16_t jpmsys5_state::coins_r(offs_t offset)
+uint16_t jpmsys5_state::coins_r(offs_t offset, uint16_t mem_mask)
 {
-	if (offset == 2)
-		return ioport("COINS")->read() << 8;
-	else
-		return 0xffff;
+	return ioport("COINS")->read() << 8;
+}
+
+uint16_t jpmsys5_state::unknown_port_r(offs_t offset, uint16_t mem_mask)
+{
+	if (m_unknown_port)
+		return m_unknown_port->read();
+
+	return 0xffff;
```

Comment:
`optional_ioport` still has `read_safe` for this.

Reply:
ah right, for some reason I thought read_safe had been removed